moshi-ir/moshi-compiler-plugin/src/main/kotlin/dev/zacsweers/moshix/ir/compiler/MoshiComponentRegistrar.kt

```diff
@@ -76,8 +77,13 @@ private fun AnalysisHandlerExtension.Companion.registerExtensionFirst(
     project: MockProject,
     extension: AnalysisHandlerExtension
 ) {
-  project
-      .extensionArea
-      .getExtensionPoint(AnalysisHandlerExtension.extensionPointName)
-      .registerExtension(extension, LoadingOrder.FIRST, project)
+  // See
+  // https://github.com/detekt/detekt/commit/a0d36e2ca4f6ca38cac0f9cb418df989ccf4f063#diff-1abfa33e705e9d1a139e397920c0a3b91cff3fe0d738291dd9bce517943290d0R24-R28
+  @Suppress("DEPRECATION")
+  synchronized(Extensions.getRootArea()) {
```

Comment:
I can't reproduce this issue but it seems a known issue and borrowing this solution from Detekt to try it out